test/e2e/pivoting_based_feature_test.go

```diff
@@ -145,3 +148,14 @@ func createTargetCluster(k8sVersion string) (targetCluster framework.ClusterProx
 	targetCluster = bootstrapClusterProxy.GetWorkloadCluster(ctx, namespace, clusterName)
 	return targetCluster
 }
+
+func deprovisionTargetCluster(client *client.Client) {
+	c := clusterv1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterName,
+			Namespace: namespace,
+		},
+	}
+	err := (*client).Delete(ctx, &c)
+	Expect(err).ToNot(HaveOccurred())
+}
```

Comment:
I think we should use [DeleteCluster](https://pkg.go.dev/sigs.k8s.io/cluster-api/test/framework#DeleteCluster) from the framework instead.